### PR TITLE
Add nginx-mod-http-perl module

### PR DESCRIPTION
**Pull request by chriscarpenter12**
Resolves issue #108 about adding support for the nginx perl module.

### CODE REVIEW

**Comment by phracek** [comment]
Thanks for the PR! Great approach.

**Comment by chriscarpenter12** [comment]
There's one thing I'm seeing that's missing with this implementation so far. The ability for users to specify their own perl modules because the only want to include files in the s2i build is through the defined config directories or the source directory which you wouldn't want these file to be publicly accessible. Working on a solution for this.

**Comment by chriscarpenter12** [comment]
Finished necessary changes to allow loading user created perl modules from the `perl_modules` and `perl_require` directives. Similar setup to the nginx config directories there's a new `nginx-perl` directory for users to place perl modules.

**Comment by phracek** [comment]
[test]

File: 1.12/test/run [hunk 1]
```diff
@@ -187,7 +186,7 @@ test_for_output()
   info "Testing URL $url with HTTP hostname ${host} produces output $expect"
 
   while [ $attempt -le $max_attempts ]; do
-    response=$(curl -s -H "Host: ${host}" ${url})
+    response=$(curl -is -H "Host: ${host}" ${url})
     echo "${response}"
     status=$?
```
Comment (phracek) [inline 1.12/test/run:191]: `$?` is return code of `echo` command.  Is this intended?

File: 1.14/test/run [hunk 2]
```diff
@@ -187,7 +186,7 @@ test_for_output()
   info "Testing URL $url with HTTP hostname ${host} produces output $expect"
 
   while [ $attempt -le $max_attempts ]; do
-    response=$(curl -s -H "Host: ${host}" ${url})
+    response=$(curl -is -H "Host: ${host}" ${url})
     echo "${response}"
     status=$?
```
Comment (phracek) [inline 1.14/test/run:191]: The same `$?` is return value of `echo` command. Is it intended?

File: 1.16/test/run [hunk 3]
```diff
@@ -187,7 +186,7 @@ test_for_output()
   info "Testing URL $url with HTTP hostname ${host} produces output $expect"
 
   while [ $attempt -le $max_attempts ]; do
-    response=$(curl -s -H "Host: ${host}" ${url})
+    response=$(curl -is -H "Host: ${host}" ${url})
     echo "${response}"
     status=$?
```
Comment (phracek) [inline 1.16/test/run:191]: The same as above

File: 1.12/test/run (hunk 1 above)
Comment (chriscarpenter12) [reply 1.12/test/run:191]: Great catch! I didn't even see that. Will updated in a minute.

**Comment by chriscarpenter12** [comment]
> Only minor issues.

Cleaned up those order issues and reran the tests. All passed locally.

**Comment by phracek** [comment]
[test][test-openshift]

**Comment by chriscarpenter12** [comment]
@phracek What is the usual timeframe for merges to hit a release to redhat registry?

**Comment by notroj** [comment]
Thanks a lot @chriscarpenter12 for the contribution!

**Comment by phracek** [comment]
@chriscarpenter12 Approximately every 6 weeks, as far as I know, but It can be a little bit more.

**Comment by chriscarpenter12** [comment]
According to the tags on that container one was released 3 days ago. Do we have to wait 6 weeks then? Was hoping to get these changes :)

**Pull request merged**